hydro_deploy/core/src/azure.rs

```diff
@@ -128,9 +128,10 @@ impl Host for AzureHost {
     }
 
     fn collect_resources(&self, resource_batch: &mut ResourceBatch) {
-        if self.launched.get().is_some() {
-            return;
-        }
+        assert!(
```

Comment:
I don't think this will do the right thing; we want deploys to be idempotent, which is why we just return when we already have a machine.

Reply:
Alrighty, reverted